scipy-stubs/signal/_peak_finding.pyi

```diff
@@ -1,37 +1,86 @@
-from scipy._typing import Untyped
+from collections.abc import Callable, Sequence
+from typing import Concatenate, Literal, TypeAlias, TypedDict
+
+import numpy as np
+import numpy.typing as npt
+import optype as op
+from numpy._typing import _ArrayLikeFloat_co, _ArrayLikeInt_co
+from scipy._typing import AnyInt, AnyReal
 
 __all__ = ["argrelextrema", "argrelmax", "argrelmin", "find_peaks", "find_peaks_cwt", "peak_prominences", "peak_widths"]
 
-def argrelmin(data: Untyped, axis: int = 0, order: int = 1, mode: str = "clip") -> Untyped: ...
-def argrelmax(data: Untyped, axis: int = 0, order: int = 1, mode: str = "clip") -> Untyped: ...
-def argrelextrema(data: Untyped, comparator: Untyped, axis: int = 0, order: int = 1, mode: str = "clip") -> Untyped: ...
-def peak_prominences(x: Untyped, peaks: Untyped, wlen: Untyped | None = None) -> Untyped: ...
+_Array_n: TypeAlias = npt.NDArray[np.intp]
+_Array_n_1d: TypeAlias = np.ndarray[tuple[int], np.dtype[np.intp]]
+_Array_f8: TypeAlias = npt.NDArray[np.float64]
+_Mode: TypeAlias = Literal["clip", "wrap"]
+
+_ProminencesResult: TypeAlias = tuple[_Array_f8, _Array_n, _Array_n]
+_WidthsResult: TypeAlias = tuple[_Array_f8, _Array_f8, _Array_f8, _Array_f8]
+
+class _FindPeaksResultsDict(TypedDict, total=False):
+    peak_heights: _Array_f8
+    left_thresholds: _Array_f8
+    right_thresholds: _Array_f8
+    prominences: _Array_f8
+    left_bases: _Array_n
+    right_bases: _Array_n
+    width_heights: _Array_f8
+    left_ips: _Array_f8
+    right_ips: _Array_f8
+    plateau_sizes: _Array_n
+    left_edges: _Array_n
+    right_edges: _Array_n
+
+def argrelmin(
+    data: npt.NDArray[np.generic],
+    axis: op.CanIndex = 0,
+    order: int = 1,
```

Comment:
I think that besides `int`, you could also pass some `np.integer[Any]` and `np.bool_` to `order`. And coincidentally, `scipy._typing.AnyInt` is an alias for exactly that :)

There's also some other `order`s like this below